### PR TITLE
some fixes

### LINKED ISSUE #992: [1.12.2]IFoodStats.saturationLevel can't be given value
## Issue Description
Internal Server Exception after setting a value for an IFoodStats

#### What you expected to happen:
Normally set a saturation level for a player

#### Script used (Please Pastebin or gist your script, posting an unpasted or ungist'd script will automatically close this issue):

[https://github.com/ET-Team/EnigTech2/blob/master/EnigTech2/minecraft/scripts/contenttweaker/item/AdvancedFruit.zs#L33](url)

#### crafttweaker.log file (Please Pastebin or gist your file, posting an unpasted or ungist'd file will automatically close this issue):

[crafttweaker.log](https://github.com/CraftTweaker/CraftTweaker/files/4759396/crafttweaker.log) (Too big for pastebin)
note: No [ERROR] present.
____
#### Affected Versions (Do *not* use "latest"):

- Minecraft:1.12.2
- Forge:2854
- Crafttweaker: CraftTweaker2-1.12-4.1.20.574
- Using a server: yes
- If yes, does the client have the exact same scripts? yes

#### Your most recent log file where the issue was present: 

https://paste.ubuntu.com/p/64MfHpdZFf/
Line 4433

### PR DESCRIPTION
**Pull request by friendlyhj**
Close #992

### CODE REVIEW

File: CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataByte.java [hunk 1]
```diff
@@ -17,7 +17,7 @@ public DataByte(byte value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a byte to a bool");
+        return value != 0;
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataByte.java]: I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1`

File: CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataDouble.java [hunk 2]
```diff
@@ -17,7 +17,7 @@ public DataDouble(double value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a double to a bool");
+        return value != 0.0;
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataDouble.java]: I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1.0`

File: CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataFloat.java [hunk 3]
```diff
@@ -17,7 +17,7 @@ public DataFloat(float value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a float to a bool");
+        return value != 0.0f;
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataFloat.java]: I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1.0f`

File: CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataInt.java [hunk 4]
```diff
@@ -17,7 +17,7 @@ public DataInt(int value) {
     
     @Override
     public boolean asBool() {
-        throw new RuntimeException("Cannot cast an int to a bool");
+        return value != 0;
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataInt.java]: I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1`

File: CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataLong.java [hunk 5]
```diff
@@ -17,7 +17,7 @@ public DataLong(long value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a long to a bool");
+        return value != 0;
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataLong.java]: I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1`

File: CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataShort.java [hunk 6]
```diff
@@ -17,7 +17,7 @@ public DataShort(short value) {
     
     @Override
     public boolean asBool() {
-        throw new IllegalDataException("Cannot cast a short to a bool");
+        return value != 0;
```
Comment (jaredlll08) [inline CraftTweaker2-API/src/main/java/crafttweaker/api/data/DataShort.java]: I feel this should better mimic how Java does it, which is "if this isn't true (or 1), then it should be considered false).

So change this to `value == 1`

File: CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/commands/NBTUtils.java [hunk 7]
```diff
@@ -119,14 +119,14 @@ public static String getAppealingString(String string) {
...
-        s.append("\n\u00A7e├");
+        s.append("\n\u00A7e\u251c");
         for(int j = 0; j < indent; j++) {
             s.append("\u00A7e    ");
         }
     }
     
     private static void addNewLineNoColor(StringBuilder s, int indent) {
-        s.append("\n├");
+        s.append("\n\u251c");
```
Comment (jaredlll08) [inline CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/commands/NBTUtils.java:129]: What does this do?

File: CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCFoodStats.java [hunk 8]
```diff
@@ -60,6 +60,6 @@ public float getSaturationLevel() {
     
     @Override
     public void setSaturationLevel(float saturationLevel) {
-        foodStats.setFoodSaturationLevel(saturationLevel);
+        foodStats.addExhaustion(saturationLevel - getSaturationLevel());
```
Comment (jaredlll08) [inline CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCFoodStats.java:63]: Why this change? Didn't it do what it meant to do before?

**Review by jaredlll08 — CHANGES_REQUESTED** [review]

File: CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCFoodStats.java (hunk 8 above)
Comment (friendlyhj) [reply CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/player/MCFoodStats.java:63]: the `setFoodSaturationLevel` method is `@SideOnly(CLIENT)` so calling it on server side would get a crash.

File: CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/commands/NBTUtils.java (hunk 7 above)
Comment (friendlyhj) [reply CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/commands/NBTUtils.java:129]: Due to encoding issue on my computer, the character make the my compile work fail. Yeah, it is issue on my side, I shouldn't pushed there.

Comment (friendlyhj) [reply CraftTweaker2-MC1120-Main/src/main/java/crafttweaker/mc1120/commands/NBTUtils.java:129]: btw, the unicode is `├`. Leaving it don't change anything.

**Pull request merged**